spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -121,32 +113,36 @@ protected RewriteDataFilesSparkAction self() {
   @Override
   public RewriteDataFilesSparkAction binPack() {
     Preconditions.checkArgument(
-        rewriter == null, "Must use only one rewriter type (bin-pack, sort, zorder)");
-    this.rewriter = new SparkBinPackDataRewriter(spark(), table);
+        runner == null, "Must use only one rewriter type (bin-pack, sort, zorder)");
```

Comment:
Follow up, We need to clean up these messages. They probably should say, "Rewriter type already set to %s" or something like that

Reply:
This is a second request to fix the error message, and it is not too complicated. So changed.

The solution is a bit lame:
```
    Preconditions.checkArgument(
        runner == null,
        "Rewriter type already set to %s",
        runner == null ? null : runner.description());
```
We need a second null check for the error message, or a null check around it 😢 
Decided to hide this ugliness in a method.
If you have better ideas, feel free to comment